operator/cnp_event.go

```diff
@@ -91,11 +132,50 @@ func enableCNPWatcher() error {
 	controller.NewManager().UpdateController("cnp-to-groups",
 		controller.ControllerParams{
 			DoFunc: func(ctx context.Context) error {
-				groups.UpdateCNPInformation()
+				// Retrieves all the CNP that has currently a
+				// derivative policy and creates the new
+				// derivatives policies with the latest
+				// information from providers.  To avoid issues
+				// with rate-limiting this function will
+				// execute the addDerivative function with a
+				// max number of concurrent calls, defined on
+				// maxConcurrentUpdates.
+				cnpToUpdate := cnpCache.GetAllCNP()
+				sem := make(chan bool, maxConcurrentUpdates)
+				for _, cnp := range cnpToUpdate {
+					sem <- true
+					go func(cnp *cilium_v2.CiliumNetworkPolicy) {
+						defer func() { <-sem }()
+						addDerivativeCNP(cnp)
```

Comment:
Are we blindly creating the derivative of a CNP or are we only updating the derivative CNP based on the difference in the information from providers?

Reply:
At this point we are updating derivatives just in case provider information has changed (it's mostly for aws `toGroups`).